test-skripsi/test-1.py

```diff
@@ -0,0 +1,41 @@
+
+def calculate_discount(price, discount):
+    discounted_price = price * discount
+    return discounted_price
+
+def factorial(n):
+    if n == 1:
+        return 1
+    else:
+        return n * factorial(n)  
+
+def parse_user_input():
+    try:
+        user_input = input("Enter a number: ")
+        number = int(user_input)
+    except ValueError:
+        print("Invalid input! Please enter a valid number.")
+        return None
+    return number
+
+def fibonacci(n):
+    fib_sequence = [0, 1]
+    if n < 0:
+        print("Fibonacci number cannot be calculated for negative numbers")
+        return None
+    elif n <= len(fib_sequence):
```

Comment:
The condition `n <= len(fib_sequence)` is incorrect for determining if the Fibonacci number can be returned from `fib_sequence`. Since `fib_sequence` starts with two elements `[0, 1]`, the condition should be `n <= len(fib_sequence) - 1` to correctly index into `fib_sequence`.